src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/DataFrameTests.cs

```diff
@@ -691,5 +691,14 @@ public void TestSignaturesV2_4_X()
                 Assert.IsType<RelationalGroupedDataset>(df.Pivot(Col("age"), values));
             }
         }
+
+        /// <summary>
+        /// Test signatures for APIs introduced in Spark 3.*
+        /// </summary>
+        [SkipIfSparkVersionIsLessThan(Versions.V3_0_0)]
+        public void TestSignaturesV3_X_X()
+        {
+            Assert.IsType<Row[]>(_df.ToLocalIterator(true).ToArray());

```

Comment:
Since we added more logic other than simply calling the function on the JVM side, can you add a correctness test as well?